lib/rubygems/request_set.rb

```diff
@@ -157,13 +162,11 @@ def install options, &block # :yields: request, installer
         end
       end
 
-      path = req.download cache_dir
-
-      inst = Gem::Installer.at path, options
-
-      yield req, inst if block_given?
+      spec = req.spec.install options do |installer|
+        yield req, installer if block_given?
+      end
```

Comment:
this fixes:
```
    no implicit conversion of nil into String
        /home/mpapis/.rvm/rubies/ruby-2.2.2/lib64/ruby/site_ruby/2.2.0/rubygems/package/file_source.rb:21:in `exist?'
        /home/mpapis/.rvm/rubies/ruby-2.2.2/lib64/ruby/site_ruby/2.2.0/rubygems/package/file_source.rb:21:in `present?'
        /home/mpapis/.rvm/rubies/ruby-2.2.2/lib64/ruby/site_ruby/2.2.0/rubygems/package.rb:136:in `new'
        /home/mpapis/.rvm/rubies/ruby-2.2.2/lib64/ruby/site_ruby/2.2.0/rubygems/installer.rb:102:in `at'
        /home/mpapis/.rvm/rubies/ruby-2.2.2/lib64/ruby/site_ruby/2.2.0/rubygems/request_set.rb:167:in `block in install'
        /home/mpapis/.rvm/rubies/ruby-2.2.2/lib64/ruby/site_ruby/2.2.0/rubygems/request_set.rb:156:in `each'
        /home/mpapis/.rvm/rubies/ruby-2.2.2/lib64/ruby/site_ruby/2.2.0/rubygems/request_set.rb:156:in `install'
```
which was happening because of git gems returning `nil` in `Gem::Source::Git.download`